pkg/storage/ec/client.go

```diff
@@ -203,3 +207,24 @@ func closeConn(ps client.PSClient, nodeID string) {
 		zap.S().Errorf("Failed closing connection to node %s: %v", nodeID, err)
 	}
 }
+
+func unique(nodes []*proto.Node) bool {
+	if len(nodes) < 2 {
+		return true
+	}
+
+	ids := make([]string, len(nodes))
+	for i, n := range nodes {
+		ids[i] = n.GetId()
+	}
+
+	// sort the ids and check for identical neighbors
+	sort.Strings(ids)
```

Comment:
this works. another way is to use a map

Reply:
True. I just decided to go with the solution that is easiest to read. I was pretty disappointed there isn't any function for this in the standard Go packages.

Reply:
i agree it's a bummer the stdlib doesn't have something more nice for this, but the following is a bit shorter:

```
  seen := map[string]bool
  for _, n := range nodes {
    if seen[n.Id] {
      return false
    }
    seen[n.Id] = true
  }
  return true
```

Reply:
Yes, it looks better. Perhaps, we should have a utility package with such gems.